src/mmwtex.c

```diff
@@ -5274,51 +5274,49 @@ flag writeBibliography(vstring bibFile,
           /* **IMPORTANT** Make sure to update mmhlpa.c HELP WRITE BIBLIOGRAPHY
              if new items are added to this list. */
           if (0
-              /* Put the most frequent ones first to speed up search;
-                 TODO: count occurrences in mmbiblio.html to find optimal order */
-              || !strcmp(mid(str2, k, (long)strlen("THEOREM")), "THEOREM")
```

Comment:
This list is searched sequentially.  Matching an item at the beginning is obviously way more efficient (1 comparison) than one at the end (40 comparisons).  So if you know, the keyword in str2 is 'theorem' 70% of the time, it makes some sense, to have it at the beginning.  This is the gist of the deleted comment here.  I think, the first 5 elements of the old list were deliberately put in the early position to speed up things.  The tail starting with 'AXIOM' are matched fairly seldom, so their exact position does not really matter, and we can keep an alphabetic order there, without compromising speed to a noticeable amount.

What do I personally think of it?  Well, if search time really matters, than there are better strategies around, like binary search, for example.  This is a poor man's optimization.  It may have an improving effect, though, depending on how often a search is called.  Without knowing details, I would keep the first five elements, but order the tail.

The 0 in the beginning of the list is likely optimized out and just there to keep all items, including the first, in the same textual format.

Reply:
From a technical point of view, I do not object to this PR any more, unless you see a slow down due to the impoverished search.

Reply:
I do not know C well enough, so I rely on you here. What order do you prefer ? Is the following ok (including comments) ?
```
           /* **IMPORTANT** Make sure to update mmhlpa.c HELP WRITE BIBLIOGRAPHY
              if new items are added to this list. */
           if (0
              /* The first five keywords are more frequent so are put first for
                  efficiency; the rest is in alphabetical order. */
              || !strcmp(mid(str2, k, (long)strlen("THEOREM")), "THEOREM")
              || !strcmp(mid(str2, k, (long)strlen("EQUATION")), "EQUATION")
              || !strcmp(mid(str2, k, (long)strlen("DEFINITION")), "DEFINITION")
              || !strcmp(mid(str2, k, (long)strlen("LEMMA")), "LEMMA")
              || !strcmp(mid(str2, k, (long)strlen("EXERCISE")), "EXERCISE")
              // ---- end of optimized search -----
              ... [alphabetical]
```
?

Reply:
This is fine

Reply:
Thanks. That's the content of my latest commit. So I guess you can approve. (On this repository, I have no right to approve or merge --- and am not asking for one.)